server/middleware/auth.js

```diff
@@ -0,0 +1,23 @@
+const jwt = require("jsonwebtoken");
+
+const auth = (req, res, next) => {
+  // validate if token is present & get token
+  const cookieName = process.env.COOKIE_NAME || "token";
+  const token =
+    req.headers.cookie && req.headers.cookie.split(`${cookieName}=`)[1];
+  if (!token) return res.status(401).send("Not Authorized");
+  const secret = process.env.JWT_KEY || "secret";
+  try {
+    // verify token - user id and email
+    const { id, email } = jwt.verify(token, secret);
+    req.user = {
+      id,
+      email,
+    };
```

Comment:
This is good, by doing this though, people need to realize that they need to make API call to get the user in their routes 👍

Reply:
@skassam21 This feels like a security v. performance issue. Should we opt for the additional API call and assume that at some point we could cache recent users to reduce the overhead?

Reply:
Yeah, I was actually just thinking about this - most of the time you will need the user object so yes - I'd prefer that we actually make the additional API call here. Then we can also just return a 4xx if the user doesn't exist anymore for any reason.